server/routerlicious/packages/routerlicious-base/src/nexus/runnerFactory.ts

```diff
@@ -56,31 +60,163 @@ export class OrdererManager implements core.IOrdererManager {
 	) {}
 
 	public async getOrderer(tenantId: string, documentId: string): Promise<core.IOrderer> {
-		if (!this.globalDbEnabled) {
-			const messageMetaData = { documentId, tenantId };
-			Lumberjack.info(`Global db is disabled, checking orderer URL`, messageMetaData);
-			const tenant = await this.tenantManager.getTenant(tenantId, documentId);
-
-			Lumberjack.info(
-				`tenant orderer: ${JSON.stringify(tenant.orderer)}`,
-				getLumberBaseProperties(documentId, tenantId),
+		const ordererConnectionType = await this.getOrdererConnectionType(tenantId, documentId);
+		const ordererId = this.getOrdererConnectionMapKey(tenantId, documentId);
+		if (this.ordererConnectionCloseTimeoutMap.has(ordererId)) {
+			// Clear any existing connection timeout because a new connection was added.
+			clearTimeout(this.ordererConnectionCloseTimeoutMap.get(ordererId));
+		}
+
+		if (ordererConnectionType === "local") {
+			Lumberjack.info(`Using local orderer`, getLumberBaseProperties(documentId, tenantId));
+			return this.localOrderManager.get(tenantId, documentId).then((orderer) => {
+				this.updateOrdererConnectionCount(tenantId, documentId, "increment", "local").catch(
+					(error) => {
+						Lumberjack.error(
+							`Failed to update local orderer connection count`,
+							{
+								documentId,
+								tenantId,
+							},
+							error,
+						);
+					},
+				);
+				return orderer;
+			});
+		}
+
+		Lumberjack.info(`Using Kafka orderer`, getLumberBaseProperties(documentId, tenantId));
+		return this.kafkaFactory.create(tenantId, documentId).then((orderer) => {
```

Comment:
Similar to above comment, catch() might be needed for this as well.

Reply:
Technically, this promise is being returned, so any error will be bubbled up as a Promise rejection. There is no need to catch it in this function.

Regardless, switched to `await` to make that more clear.